dev/Repeater/ViewportManagerWithPlatformFeatures.cpp

```diff
@@ -264,6 +279,35 @@ void ViewportManagerWithPlatformFeatures::OnEffectiveViewportChanged(winrt::Fram
     REPEATER_TRACE_INFO(L"%ls: \tEffectiveViewportChanged event callback \n", GetLayoutId().data());
     UpdateViewport(args.EffectiveViewport());
 
+    // As long as the make_anchor element is within the visible window, force it as the anchor.
+    // As soon as it goes outside, stop forcing it to be the anchor and
+    // let the anchor provider pick a new anchor.
+    if (auto makeAnchorElement = m_makeAnchorElement.get())
+    {
+        auto visibleWindow = GetLayoutVisibleWindowDiscardAnchor();
+        auto layoutSlot = winrt::LayoutInformation::GetLayoutSlot(makeAnchorElement.as<winrt::FrameworkElement>());
+        auto adjustedLayoutSlot = winrt::Rect(layoutSlot.X + m_layoutExtent.X, layoutSlot.Y + m_layoutExtent.Y, layoutSlot.Width, layoutSlot.Height);
+        bool doesNotOverlap = (adjustedLayoutSlot.X + adjustedLayoutSlot.Width) < visibleWindow.X ||
```

Comment:
Is there some standard rect intersection method you can call to get what you want here?